osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorPlayer.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps;
+using osu.Game.Scoring;
+using osu.Game.Screens.OnlinePlay.Multiplayer.Spectate.Sync;
+using osu.Game.Screens.Play;
+
+namespace osu.Game.Screens.OnlinePlay.Multiplayer.Spectate
+{
+    /// <summary>
+    /// A single spectated player within a <see cref="MultiSpectatorScreen"/>.
+    /// </summary>
+    public class MultiSpectatorPlayer : SpectatorPlayer
+    {
+        private readonly Bindable<bool> waitingOnFrames = new Bindable<bool>(true);
+        private readonly Score score;
+        private readonly ISpectatorPlayerClock spectatorPlayerClock;
+
+        /// <summary>
+        /// Creates a new <see cref="MultiSpectatorPlayer"/>.
+        /// </summary>
+        /// <param name="score">The score containing the player's replay.</param>
+        /// <param name="spectatorPlayerClock">The clock controlling the gameplay running state.</param>
+        public MultiSpectatorPlayer([NotNull] Score score, [NotNull] ISpectatorPlayerClock spectatorPlayerClock)
```

Comment:
Wondering if we should be passing the master clock down to the player and having it create its client clock. The main reason I bring this up is the weirdness of binding to the property of something coming in via the `ctor` (and the fact that this is the only case of binding.. at which point maybe ownership makes more sense?).

Reply:
Players are late initialised, whenever they decide to say they've begun gameplay / that message is received from the spectator server. So I think this is the best that can be done, otherwise the sync manager will assume all players are loaded because their clocks haven't been added yet?

Reply:
I see. Is that done with intent? ie. is there any reason we don't just load them all as soon as the screen loads then pause on the pre-gameplay phase (not saying we necessarily should, just curious why it is done this way).

Reply:
What do you load? Which beatmap/mods/ruleset? My primary thought going into it is that this supports that as is due to the nature of `SpectatorScreen`.

Otherwise, I could use the `PlaylistItem` for the beatmap/ruleset and extract the mods from each player (and leave the per-player beatmap/ruleset support until it's supported for multi in general), but then I also have to answer the question of how do I integrate that into `SpectatorScreen`? Sounds like at that point `SpectatorScreen` should basically not be overridden because it's a completely different flow of events.

Reply:
You're correct, but I don't think multiplayer should be a special case. That's an issue with spectator in general.

Reply:
So would you see the data flow (as you described just above) being feasible to change at the `SpectatorScreen` level without too much fuss?

Reply:
Not the `SpectatorScreen` data flow. If multiplayer should be a special case then it basically cannot use `SpectatorScreen` due to the complexity.

The problem is the clients being spectated only send their gameplay started event in `LoadComplete`:
https://github.com/ppy/osu/blob/72d9cdcdf919a9ed4032eeb99742e5783e5781d4/osu.Game/Rulesets/UI/ReplayRecorder.cs#L46-L53

Reply:
Hmm.. so you're saying the way forward you see is to start loading the player when they player starts loading, rather than when they start sending frames? And making spectator also do that?

I'm mostly thinking about how we would want a tournament stream to look going forward, and whether this aligns with that vision, if that gives context to why I'm bringing this up.

Reply:
As for tournaments, wouldn't it make more sense to always start gameplay at 0? At which point this becomes a problem of making sure that we've received all earlier frames (which I'm pretty sure doesn't hapen if we start spectating during gameplay, for example).

> Hmm.. so you're saying the way forward you see is to start loading the player when they player starts loading, rather than when they start sending frames? And making spectator also do that?

Exactly.

Reply:
I'm going to need a bit of time to think about this. I was hoping we'd be able to provide a slightly less "flashy" view in the case where all clients are loaded and playing, but maybe as you say is fine.

As for "always starting from 0", we also need to consider the case of starting spectating from mid-game, which should definitely be possible (consider a scenario where lazer crashes for whatever reason and you want to pick back up on watching an ongoing match).

Reply:
Maybe it's a matter of providing a button to go to the current time? I could imagine this being useful in normal spectator too, with its rewind functionality.

Reply:
Probably should default to current time, but sure.

Also brings up a point of when seeking support will be added to this screen (should be added soon after implementation as it feels like one of the minimum requirements for the feature).